app/models/custom_filter.rb

```diff
@@ -70,8 +70,8 @@ def self.cached_filters_for(account_id)
       scope.to_a.group_by(&:custom_filter).each do |filter, keywords|
         keywords.map! do |keyword|
           if keyword.whole_word
-            sb = /\A[[:word:]]/.match?(keyword.keyword) ? '\b' : ''
-            eb = /[[:word:]]\z/.match?(keyword.keyword) ? '\b' : ''
+            sb = /\A|^[[:word:]]/.match?(keyword.keyword) ? '\b' : ''
+            eb = /[[:word:]]\z|$/.match?(keyword.keyword) ? '\b' : ''
```

Comment:
Whilst this does work, we may actually want to use the logic that the streaming server uses here, which is "If the filter starts with or end with non-whitespace, do a boundary match"

```suggestion
            sb = /^[\S]/.match?(keyword.keyword) ? '\b' : ''
            eb = /[\S]$/.match?(keyword.keyword) ? '\b' : ''
```

It seems ruby's \A and \z do not match symbols, only alphanumerical characters.